.eslintrc.json

```diff
@@ -0,0 +1,32 @@
+{
+    "env": {
+        "node": true,
+        "es6": true,
+        "jest": true
+    },
+    "extends": [ "eslint:recommended" ],
+    "rules": {
+        "indent": [
+            "error",
+            2
+        ],
+        "linebreak-style": [
```

Comment:
Is there any reference showing `.gitattributes` is the legitimate alternative for linter new line check?

Reply:
Github desktop is not the standard of software development. I mean some solid reference that despite what Github do, recoment to use gitattribute instead of linter check. I think gitattribute in this case is used to make sure windows users wont mess up the git history by their CRLF. It doesn't mean to remove linter as it is the QA standard.

Let's say you assign a person in the manufacturing company to make sure bad things are taken of boxes. This doesn't mean to go and remove your QA standards for removing those bad things from boxes. Actually if we add that line in gitattribute it is because we have this QA condition of checking new line character.

Isn't it?

Reply:
I just saw this part in the official documentation:

![image](https://user-images.githubusercontent.com/11241315/153563293-b4d4b054-0a62-42ea-98ae-3d29e902debf.png)

> When text is set to "auto", the path is marked for automatic end-of-line conversion. If Git decides that the content is text, its line endings are converted to LF on checkin. When the file has been committed with CRLF, no conversion is done.

As per my understanding, if there is a file committed with CRLF, new commits will be made with CRLF. This is not what we want and this is why we still need to have QA mechanism to make sure any manual/automatic approach ends in having only LF.

Reply:
I just added `.gitattribite` file with this line: `* eol=lf` to make sure eol is set to LF all the time. Good?

Reply:
@ahangarha I guess that means if a file is already committed with CRLF, it won't be conveted to LF if you add `text=auto` at a later time. That's why it's a good idea to add it once you create a repo.

I don't see that as an issue since I'm using VS Code, I can confirm that my commits don't have CRLF.

Generally, `text=auto` is a smarter choice since it won't touch binaries, and since we don't ship binaries I don't think that `* eol=lf` would cause an issue (not sure if it would break images if we add any to the repo in the future, you should check that).

And yeah, you're right, let's keep the linter rule.

We're good to go :+1: